python/python/lance/dataset.py

```diff
@@ -2141,6 +2174,12 @@ def commit(
         # TODO: mode is never used!
         if isinstance(base_uri, Path):
             base_uri = str(base_uri)
+        elif isinstance(base_uri, LanceDataset):
+            base_uri = base_uri._ds
```

Comment:
Minor nit: maybe rename `base_uri` to `base_uri_or_dataset`?  Took me a second to realize this wasn't:

```
if isinstance(base_uri, Path):
  base_uri = str(base_uri)
elif isinstance(base_uri, LanceDataset):
  base_uri = base_uri._uri
```

Reply:
I wanted to, but there were unit tests were we passed `base_uri` as a named argument, and thus realized this could be an annoying breaking change for some users.